lib/utilities/requestUtils.js

```diff
@@ -0,0 +1,32 @@
+const { config } = require('../Config');
+
+/**
+ * getClientIp - Gets the client IP from the request
+ * @param {object} request - http request object
+ * @return {string} - returns client IP from the request
+ */
+function getClientIp(request) {
+    const clientIp = (request.connection && request.connection.remoteAddress) ||
+        (request.socket && request.socket.remoteAddress) ||
+        (request.connection && request.connection.socket &&
+        request.connection.socket.remoteAddress);
+
+    const requestConfig = config.requests;
+    if (requestConfig && requestConfig.viaProxy) {
+        /**
+         * if requests are configured to come via proxy,
+         * check from config which IPs are to be trusted and
+         * which header to be used to extract client IP
+        */
+        if (requestConfig.trustedProxyIPs.includes(clientIp)) {
```

Comment:
We'd want to test if the trusted network ranges include this ip, not if an ip list contains an ip. In dynamic environments like Kubernetes possible source ip addresses will be in the thousands.